shopelectro/views/catalog.py

```diff
@@ -229,26 +256,28 @@ def template_context(page, tag_titles, tags):
                 'tags': tags,
             }
 
-        page = context['page']
-        page.get_template_render_context = partial(
-            template_context, page, tag_titles, tags)
+        tags = self.get_tags()
```

Comment:
You are calling it two times: here and in `get_products(...)`. It will increase quantity of queries to db.

Reply:
@artemiy312 , i thought, ORM caches such queries. I'll fix it